airbyte-cdk/bulk/core/load/src/testFixtures/kotlin/io/airbyte/cdk/test/write/BasicFunctionalityIntegrationTest.kt

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.cdk.test.write
+
+import io.airbyte.cdk.command.Append
+import io.airbyte.cdk.command.ConfigurationJsonObjectBase
+import io.airbyte.cdk.command.DestinationStream
+import io.airbyte.cdk.data.ObjectTypeWithoutSchema
+import io.airbyte.cdk.message.DestinationRecord
+import io.airbyte.cdk.message.StreamCheckpoint
+import io.airbyte.cdk.test.util.DestinationCleaner
+import io.airbyte.cdk.test.util.DestinationDataDumper
+import io.airbyte.cdk.test.util.ExpectedRecordMapper
+import io.airbyte.cdk.test.util.IntegrationTest
+import io.airbyte.cdk.test.util.NameMapper
+import io.airbyte.cdk.test.util.NoopExpectedRecordMapper
+import io.airbyte.cdk.test.util.NoopNameMapper
+import io.airbyte.cdk.test.util.OutputRecord
+import io.airbyte.protocol.models.v0.AirbyteConnectionStatus
+import io.airbyte.protocol.models.v0.AirbyteMessage
+import kotlin.test.assertEquals
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.assertAll
+
+abstract class BasicFunctionalityIntegrationTest(
+    val config: ConfigurationJsonObjectBase,
+    dataDumper: DestinationDataDumper,
+    destinationCleaner: DestinationCleaner,
+    recordMangler: ExpectedRecordMapper = NoopExpectedRecordMapper,
+    nameMapper: NameMapper = NoopNameMapper,
+    /**
+     * Whether to actually verify that the connector wrote data to the destination.
+     * This should only ever be disabled for test destinations (dev-null, etc.).
+     */
+    val verifyDataWriting: Boolean = true,
+) : IntegrationTest(dataDumper, destinationCleaner, recordMangler, nameMapper) {
+    @Test
+    open fun testCheck() {
+        val process = destinationProcessFactory.createDestinationProcess("check", config = config)
+        process.run()
+        val messages = process.readMessages()
+        val checkMessages = messages.filter { it.type == AirbyteMessage.Type.CONNECTION_STATUS }
+
+        assertEquals(
+            checkMessages.size,
+            1,
+            "Expected to receive exactly one connection status message, but got ${checkMessages.size}: $checkMessages"
+        )
+        assertEquals(
+            AirbyteConnectionStatus.Status.SUCCEEDED,
+            checkMessages.first().connectionStatus.status
+        )
+    }
+
+    @Test
+    open fun testBasicWrite() {
```

Comment:
this test case is mostly here to (a) be a quick smoke test on our state handling, and (b) show the typical interaction with `DestinationProcess`. Most tests we'll actually write don't need any of the state stuff, and would look more like:
```kotlin
runSync(config, stream, listOf(records...))
dumpAndDiffRecords(...)
// (potentially more syncs, if we're e.g. testing refreshes)
```